File: GrooveOn.Models/ResponseObjects/RoleResponse.cs
using System;

namespace GrooveOn.Model.ResponseObjects
{
    public class RoleResponse
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public DateTime CreatedAt { get; set; }
        public bool IsActive { get; set; }
    }
}

File: UI/grooveon_desktop/lib/models/response/role_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'role_response.g.dart';

@JsonSerializable()
class RoleResponse {
  final int id;
  final String name;
  final String? description;
  final DateTime? createdAt;
  final bool isActive;

  RoleResponse({
    required this.id,
    required this.name,
    this.description,
    this.createdAt,
    required this.isActive,
  });

  factory RoleResponse.fromJson(Map<String, dynamic> json) =>
      _$RoleResponseFromJson(json);

  Map<String, dynamic> toJson() => _$RoleResponseToJson(this);
}
